node/src/components/storage/tests.rs

```diff
@@ -0,0 +1,698 @@
+    assert!(response.is_empty());
+}
+
+#[test]
+fn can_retrieve_store_and_load_deploys() {
+    let mut harness = ComponentHarness::default();
+    let mut storage = storage_fixture(&mut harness);
+
+    // Create a random deploy, store and load it.
+    let deploy = Box::new(Deploy::random(&mut harness.rng));
+
+    let was_new = put_deploy(&mut harness, &mut storage, deploy.clone());
+    assert!(was_new, "putting deploy should have returned `true`");
+
+    // Storing the same deploy again should work, but yield a result of `false`.
+    let was_new_second_time = put_deploy(&mut harness, &mut storage, deploy.clone());
+    assert!(
+        !was_new_second_time,
+        "storing deploy the second time should have returned `false`"
+    );
+
+    // Retrieve the stored deploy.
+    let response = get_deploys(&mut harness, &mut storage, smallvec![*deploy.id()]);
+    assert_eq!(response, vec![Some(deploy.as_ref().clone())]);
+
+    // Also ensure we can retrieve just the header.
+    let response = harness.send_request(&mut storage, |responder| {
+        StorageRequest::GetDeployHeaders {
+            deploy_hashes: smallvec![*deploy.id()],
+            responder,
+        }
+        .into()
+    });
+    assert_eq!(response, vec![Some(deploy.header().clone())]);
+
+    // Finally try to get the metadata as well. Since we did not store any, we expect empty default
+    // metadata to present.
+    let (deploy_response, metadata_response) = harness
+        .send_request(&mut storage, |responder| {
+            StorageRequest::GetDeployAndMetadata {
+                deploy_hash: *deploy.id(),
+                responder,
+            }
+            .into()
+        })
+        .expect("no deploy with metadata returned");
+
+    assert_eq!(deploy_response, *deploy);
+    assert_eq!(metadata_response, DeployMetadata::default());
+}
+
+#[test]
+fn store_and_load_a_lot_of_deploys() {
+    // There is a quite a bit confusion about whether or not `commit()` must be called after
+    // read-only transactions
+    let mut harness = ComponentHarness::default();
+    let mut storage = storage_fixture(&mut harness);
+
+    let total = 1000;
+    let batch_size = 25;
+
+    let mut deploy_hashes = Vec::new();
+
+    for _ in 0..total {
+        let deploy = Box::new(Deploy::random(&mut harness.rng));
+        deploy_hashes.push(*deploy.id());
+        put_deploy(&mut harness, &mut storage, deploy);
+    }
+
+    // Shuffle deploy hashes around to get a random order.
+    deploy_hashes.as_mut_slice().shuffle(&mut harness.rng);
+
+    // Retrieve all from storage, ensuring they are found.
+    for chunk in deploy_hashes.chunks(batch_size) {
+        let result = get_deploys(&mut harness, &mut storage, chunk.iter().cloned().collect());
+        assert!(result.iter().all(Option::is_some));
+    }
+}
+
+#[test]
+fn store_execution_results_for_two_blocks() {
+    // This is a simplified version of the larger test below.
```

Comment:
Probably worth naming the test to be explicit, or remove the comment since it's a bit superfluous?